test/eventinge2erekt/features/encryption.go

```diff
@@ -0,0 +1,202 @@
+package features
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/injection/clients/dynamicclient"
+	"knative.dev/pkg/logging"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/resources/knativeservice"
+)
+
+func VerifyEncryptedTrafficToActivatorToApp(refs []corev1.ObjectReference, since time.Time) *feature.Feature {
+	// Just a small sleep to let the logs be written.
+	time.Sleep(10 * time.Second)
+
+	f := feature.NewFeature()
+
+	f.Stable("path to activator to app").
+		Must("has encrypted traffic", func(ctx context.Context, t feature.T) {
+			env := environment.FromContext(ctx)
+
+			var (
+				ksvcName string
+				numKsvc  int
+			)
+			for _, ref := range refs {
+				if ref.GroupVersionKind().GroupVersion() == knativeservice.GVR().GroupVersion() {
+					if numKsvc != 0 {
+						t.Fatalf("Found more than one Knative Service: %s, %s", ksvcName, ref.Name)
+					}
+					ksvcName = ref.Name
+					numKsvc++
+				}
```

Comment:
This makes sense to me, I'd add a comment saying "this is for safety and guarantee the feature invariance"